### ISSUE TITLE
Merging entries

### ISSUE THREAD

**Issue by omangin** [pubs/pubs#65]
This is a discussion on ways to add more advanced behaviors for merging entries. The basic behaviors when importing an entry with an existing citekey are to either overwrite the existing one or change the citekey.

On the long run, it would be useful to be able to interactively merge entries as well as detect whether the new one is about the same paper or entity than the existing one, and to extend that approach to attached documents and notes.

One inspiration is what [beets](http://beets.io) does for music tags (see [auto-tagger](http://beets.readthedocs.io/en/v1.4.3/guides/tagger.html#using-the-auto-tagger)), which includes displaying the differences and computing a similarity score (which also enables automatic decisions). Before reaching this sophistication, a quicker move would be to simply open a diff tool (like git does) and asking interactively for overwriting docs and notes.

Any requests or thoughts on this?

**Comment by beuerle** [comment]
> On the long run, it would be useful to be able to interactively merge entries
as well as detect whether the new one is about the same paper or entity than
the existing one

As this happened to me just recently (adding a paper which I already had in the pubs library): wouldn't it be rather simple to check whether the object's DOI to be added is already present in the library (if the object has a DOI of course)? There should not be any two objects with the same DOI in the library.

**Comment by benureau** [comment]
Yes. DOIs and ISBN are rather trivial to check. There may be subtelties with versioned DOIs, but as an "helper" feature rather than a guarantees, it does adds value to `pubs`.
